ksql-serde/src/main/java/io/confluent/ksql/serde/protobuf/transformer/ProtobufTransformer.java

```diff
@@ -88,39 +91,58 @@ private Object convertField(final Field field, final MessageOrBuilder protobuf)
             throw new RuntimeException("Type mismatch?");
         }
       }
+      return convertValue(field.schema(), value);
+    }
+    return null;
+  }
 
-      // Based on the type of the field..
-      switch (field.schema().type()) {
-        case INT8:
-        case INT16:
-        case INT32:
-        case INT64:
-        case FLOAT32:
-        case FLOAT64:
-        case BOOLEAN:
-        case STRING:
-        case ARRAY:
-          return value;
-
-        case BYTES:
-          return ((ByteString)value).toByteArray();
-
-        case MAP:
-          final Map mapValue = new HashMap<>();
-          for (final MapEntry<Object, Object> mapEntry: (Collection<MapEntry>) value) {
-             mapValue.put(mapEntry.getKey(), mapEntry.getValue());
-          }
-          return mapValue;
+  @SuppressWarnings("unchecked")
+  public Object convertValue(final Schema schema, final Object value) {
+    // Based on the type of the field..
+    switch (schema.type()) {
+      case INT8:
+      case INT16:
+        return value;
+      case INT32:
+      case INT64:
+        if (value instanceof Timestamp) {
```

Comment:
I originally added this before the auto schema creation, not that it exists, might as well let them be parsed as the messages they are.